iothub/device/src/TransientFaultHandling/AsyncExecution[T].cs

```diff
@@ -37,108 +37,128 @@ internal class AsyncExecution<TResult>
     {
         internal const int MinimumTimeBetweenRetriesMiliseconds = 1000;
 
-        private readonly Func<Task<TResult>> taskFunc;
-
-        private readonly ShouldRetry shouldRetry;
-
-        private readonly Func<Exception, bool> isTransient;
-
-        private readonly Action<int, Exception, TimeSpan> onRetrying;
-
-        private readonly bool fastFirstRetry;
-
-        private readonly CancellationToken cancellationToken;
-
-        private readonly Stopwatch stopwatch;
-
-        private Task<TResult> previousTask;
-
-        private int retryCount;
-
-        public AsyncExecution(Func<Task<TResult>> taskFunc, ShouldRetry shouldRetry, Func<Exception, bool> isTransient, Action<int, Exception, TimeSpan> onRetrying, bool fastFirstRetry, CancellationToken cancellationToken)
+        protected static readonly TaskContinuationOptions s_taskContinuationOption = 
```

Comment:
`RunContinuationAsynchronously` does not exist in NET451. It will retain existing behavior.

Also note that the guidance (https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#always-create-taskcompletionsourcet-with-taskcreationoptionsruncontinuationsasynchronously) specifies a similar change (not this one) with the constructor, but doesn't say anything about this parameter to `Task.ContinueWith`. I assume we should change them both, but I'm not sure. Leaving the old value seems wrong, though.